Generator.Equals/Models/EqualityTypeModelTransformer.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+
+using Generator.Equals.Extensions;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Generator.Equals.Models;
+
+internal sealed class EqualityTypeModelTransformer
+{
+    private readonly GeneratorAttributeSyntaxContext _context;
+    private readonly CancellationToken _token;
+
+    public EqualityTypeModelTransformer(GeneratorAttributeSyntaxContext context, CancellationToken token)
+    {
+        _context = context;
+        _token = token;
+    }
+
+    public EqualityTypeModel? Transform()
+    {
+        var attributesMetadata = AttributesMetadata.Instance;
+
+        var equatableAttributeData = _context.Attributes.SingleOrDefault();
+        if (equatableAttributeData is null || !attributesMetadata.Equatable.Equals(equatableAttributeData.AttributeClass))
+        {
+            // TODO: Report diagnostic
+            // throw new Exception("Expected exactly one EquatableAttribute.");
```

Comment:
Didn't have the resources to research it but it seems like its alot of additional work since the reporting would be happening somewhere completely different

Reply:
I will reiterate that generators cannot safely report diagnostics and still be incremental. Diagnostics should be reported by a matching analyzer. 

However, I agree that silently failing is not intuitive for the user. Silently fail here and write an analyzer to report the failure.

Reply:
https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.cookbook.md#issue-diagnostics

Reply:
If you want an example of working set of Generator/Analyzer/CodeFix, please feel free to check out my source-generator packages:
* https://github.com/immediateplatform/immediate.handlers
* https://github.com/immediateplatform/immediate.apis
* https://github.com/immediateplatform/immediate.validations

Reply:
I have created an issue for this https://github.com/diegofrata/Generator.Equals/issues/65